### PR TITLE
RE2022-101: Add a minimal async SDK client

### PR DESCRIPTION
**Pull request by MrCreosote**
```python
In [1]: with open('/home/crushingismybusiness/.kbase_token_ci') as f:
   ...:     citoken = f.read().strip()
   ...:

In [2]: from src.service.sdk_async_client import SDKAsyncClient

In [3]: sdkcli = SDKAsyncClient("https://ci.kbase.us/services/ws")

In [4]: await sdkcli.call("Workspace.ver")
Out[4]: '0.14.1-dev1'

In [5]: await sdkcli.call("Workspace.get_object_info3", [{"objects": [{"ref": "6
   ...: 8953/12/1"}]}], token=citoken)
Out[5]:
{'infos': [[12,
   'lineage_correct_bacillus_A_anthracis',
   'KBaseGenomeAnnotations.Assembly-6.0',
   '2023-03-09T00:56:24+0000',
   1,
   'gaprice',
   68953,
   'gaprice:narrative_1674687340883',
   '42d1b600409f75c97333f87c51323a9c',
   51273,
   None]],
 'paths': [['68953/12/1']]}

In [6]: await sdkcli.call("Workspace.get_object_info3", [{"objects": [{"ref": "6
   ...: 8953/12/22"}]}], token=citoken)
---------------------------------------------------------------------------
ServerError                               Traceback (most recent call last)
Cell In[6], line 1
----> 1 await sdkcli.call("Workspace.get_object_info3", [{"objects": [{"ref": "68953/12/22"}]}], token=citoken)

File ~/github/kbase/collections/src/service/sdk_async_client.py:62, in SDKAsyncClient.call(self, method, params, token)
     60 err = await resp.json()
     61 if "error" in err:
---> 62     raise ServerError(**err["error"])
     63 else:
     64     raise ServerError(message=await resp.text())

ServerError: No object with id 12 (name lineage_correct_bacillus_A_anthracis) and version 22 exists in workspace 68953 (name gaprice:narrative_1674687340883)
us.kbase.workspace.database.exceptions.NoSuchObjectException: No object with id 12 (name lineage_correct_bacillus_A_anthracis) and version 22 exists in workspace 68953 (name gaprice:narrative_1674687340883)
	at us.kbase.workspace.database.mongo.MongoWorkspaceDB.resolveObjectIDs(MongoWorkspaceDB.java:3026)
*snip*

In [7]: await sdkcli.close()

In [8]: sdkcli = SDKAsyncClient("https://kbase.us")

In [9]: await sdkcli.call("Workspace.ver")
---------------------------------------------------------------------------
ClientResponseError                       Traceback (most recent call last)
Cell In[9], line 1
----> 1 await sdkcli.call("Workspace.ver")

File ~/github/kbase/collections/src/service/sdk_async_client.py:68, in SDKAsyncClient.call(self, method, params, token)
     66         raise ServerError(message=await resp.text())
     67 if not resp.ok:
---> 68     resp.raise_for_status()
     69 res = await resp.json()
     70 if "result" not in res:

File ~/.local/share/virtualenvs/collections-Ov35wl3J/lib/python3.11/site-packages/aiohttp/client_reqrep.py:1005, in ClientResponse.raise_for_status(self)
   1003 assert self.reason is not None
   1004 self.release()
-> 1005 raise ClientResponseError(
   1006     self.request_info,
   1007     self.history,
   1008     status=self.status,
   1009     message=self.reason,
   1010     headers=self.headers,
   1011 )

ClientResponseError: 403, message='Forbidden', url=URL('https://www.kbase.us/')

In [10]: await sdkcli.close()

In [11]: exit
```

### CODE REVIEW

File: src/service/app_state_data_structures.py [hunk 1]
```diff
@@ -35,14 +35,6 @@ async def get_storage(self) -> tuple[aioarango.ArangoClient, ArangoStorage]:
         """
         return await build_storage(self._cfg, self._dps)
 
-    async def get_workspace(self, token):
```
Comment (MrCreosote) [inline src/service/app_state_data_structures.py:38]: Unused, remove for now

File: src/service/sdk_async_client.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        method - the name of the method, for example `Workspace.ver`.
+        params - the parameters for the method.
+        token - the user's KBase token, if any.
+
+        Returns will be as documented in the spec for the respective service.
+        """
+        body = {
+            'method': method,
+            'params': params or [],
+            'version': '1.1',
```
Comment (Tianhao-Gu) [inline src/service/sdk_async_client.py:50]: making version a parameter and defaults to 1.1?

File: src/service/sdk_async_client.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                return
+            if len(res["result"]) == 1:
+                return res["result"][0]
+            return res["result"]
+
+    async def close(self):
+        """
+        Close the client and release resources.
+        """
+        await self._session.close()
```
Comment (Tianhao-Gu) [inline src/service/sdk_async_client.py:82]: should be fine. But I don't know if we should handle exceptions like `aiohttp.ClientConnectionError` during closing the session.

**Review by Tianhao-Gu — APPROVED** [review]
👍

File: src/service/sdk_async_client.py (hunk 2 above)
Comment (MrCreosote) [reply src/service/sdk_async_client.py:50]: A couple reasons I'm not sure about this:
* Why would the user ever care which version of the JSON-RPC protocol is used? It should be invisible to them either way
* The client is ported from the SDK `baseclient.py` and I have no idea if the code even works for JSON-RPC 2.0.

File: src/service/sdk_async_client.py (hunk 3 above)
Comment (MrCreosote) [reply src/service/sdk_async_client.py:82]: I'm not sure what we'd do if an exception was thrown. Catch it and then do... what?

**Pull request merged**